vignettes/datatable-reference-semantics.Rmd

```diff
@@ -112,14 +112,33 @@ Note that the code above explains how `:=` can be used. They are not working exa
 
 * On the other hand, (b) is handy if you would like to jot some comments down for later.
 
-* In general, (a) and (b) should produce the same result, however there are minor differences in certain cases due to the implementation details of `:=`. Specifically, when using (b), the `:=` operator behaves like an alias to `list`, therefore if `RHS` is a `list` in functional form the column will become a list of lists.
+* In general, (a) and (b) should produce the same result, however there are minor differences in certain cases due to the implementation details of `:=`. Specifically, when using (b), the `:=` operator behaves like an alias to `list`, therefore if `RHS` is a `list` in functional form the column will become a list of lists. See example below.
 
 * The result is returned *invisibly*.
 
 * Since `:=` is available in `j`, we can combine it with `i` and `by` operations just like the aggregation operations we saw in the previous vignette.
 
 #
 
+Although in most cases they are the same, there is a minor difference between the standard and functional forms mentioned above. Let's see an example to understand this.
+```{r eval = FALSE}
+DT = data.table(a = list('A', 'B', 'C'))
+l = list(1L:3L)
+
+DT[, new := l] # Standard form, new column is a list of integers
```

Comment:
please change to "new column is integer" (remove list as that is confusing, it is not a list column)
also maybe change column name from new to new_int